### ISSUE TITLE
ocamlfind won't build with ocamljava

### ISSUE THREAD

**Issue by arlencox** [xclerc/ocamljava#26]
When I try to build ocamlfind using the ocaml built with ocamljava I get an inconsistent assumptions over interface errors:

```
File "topfind.ml", line 1:
# Error: The files /Users/cox/.opam/ocamljava-2.0-alpha3/lib/ocaml/compiler-libs/toploop.cmi
#        and /Users/cox/.opam/ocamljava-2.0-alpha3/lib/ocaml/compiler-libs/parsetree.cmi
#        make inconsistent assumptions over interface Warnings
# make[1]: *** [topfind.cmo] Error 2
```

I tried to figure out how to fix it, but I couldn't figure out how to make the pieces of the build work outside of `build.sh`

**Comment by xclerc** [comment]
Indeed; ocamlfind has to be patched to be compiled with `ocamljava`.
There is a dedicated opam package, namely `ocj-ocamlfind`.

Did you try to install ocamlfind thourgh `opam install ocj-ocamlfind`?

**Comment by arlencox** [comment]
I didn't.  I will try that.  Is there a reason why these interfaces cannot make consistent assumptions?

**Comment by xclerc** [comment]
The problem is that `ocamljava` defines new warnings.

In order to have a uniform handling, these new warnings are added as
new cases to the sum type of _legacy_ warnings, hence modifying the
signature of the `Warnings` module.
